pkg/v1/mutate/oci.go

```diff
@@ -0,0 +1,113 @@
+package mutate
+
+import (
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/match"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+)
+
+// Drops docker specific properties
+// See: https://github.com/opencontainers/image-spec/blob/main/config.md
+func toOCIV1Config(config v1.Config) v1.Config {
+	return v1.Config{
+		User:         config.User,
+		ExposedPorts: config.ExposedPorts,
+		Env:          config.Env,
+		Entrypoint:   config.Entrypoint,
+		Cmd:          config.Cmd,
+		Volumes:      config.Volumes,
+		WorkingDir:   config.WorkingDir,
+		Labels:       config.Labels,
+		StopSignal:   config.StopSignal,
+	}
+}
+
+func toOCIV1ConfigFile(cf *v1.ConfigFile) *v1.ConfigFile {
+	return &v1.ConfigFile{
+		Created:      cf.Created,
+		Author:       cf.Author,
+		Architecture: cf.Architecture,
+		OS:           cf.OS,
+		OSVersion:    cf.OSVersion,
+		History:      cf.History,
+		RootFS:       cf.RootFS,
+		Config:       toOCIV1Config(cf.Config),
+	}
+}
+
+// OCIImage mutates the provided v1.Image to be OCI compilant v1.Image
+// Check image-spec to see which properties are ported and which are dropped.
+// https://github.com/opencontainers/image-spec/blob/main/config.md
+func OCIImage(base v1.Image) (v1.Image, error) {
+	m, err := base.Manifest()
+	if err != nil {
+		return nil, err
+	}
+
+	manifest := m.DeepCopy()
+
+	for i, layer := range manifest.Layers {
+		switch layer.MediaType {
+		case types.DockerLayer:
+			manifest.Layers[i].MediaType = types.OCILayer
+		case types.DockerUncompressedLayer:
+			manifest.Layers[i].MediaType = types.OCIUncompressedLayer
```

Comment:
I did not cover the other cases as they don't have their OCI counterparts. I just let arbitrary descriptors (docker-specific and user-defined ones) pass through. docker hub was pretty happy accepting a mix of these. 

these two left unchanged: `DockerPluginConfig` and `DockerForeignLayer`